src/test/java/org/folio/rs/service/RetrievalQueueServiceUnitTest.java

```diff
@@ -0,0 +1,187 @@
+package org.folio.rs.service;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Collections;
+import javax.persistence.EntityNotFoundException;
+import org.folio.rs.client.InventoryClient;
+import org.folio.rs.client.LocationClient;
+import org.folio.rs.client.UsersClient;
+import org.folio.rs.domain.dto.Contributor;
+import org.folio.rs.domain.dto.EffectiveCallNumberComponents;
+import org.folio.rs.domain.dto.EffectiveLocation;
+import org.folio.rs.domain.dto.FolioLocation;
+import org.folio.rs.domain.dto.Item;
+import org.folio.rs.domain.dto.LocationMapping;
+import org.folio.rs.domain.dto.MovedEvent;
+import org.folio.rs.domain.dto.MovedEventRequest;
+import org.folio.rs.domain.dto.ResultList;
+import org.folio.rs.domain.dto.User;
+import org.folio.rs.domain.entity.RetrievalQueueRecord;
+import org.folio.rs.mapper.MovedEventMapper;
+import org.folio.rs.repository.RetrievalQueueRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class RetrievalQueueServiceUnitTest {
```

Comment:
What especial do we test here in comparison with RetrievalQueueServiceTest? Why we cannot to have one test class for all the Retrieval Logic?

Reply:
RetrievalQueueServiceTest checks saving of RetrievalQueueRecord to db and its extracting. RetrievalQueueServiceUnitTest test pubsub integration, so we need responses from users,inventory clients etc, and we are using mocks for that.Maybe, we should rename this class to better display its purpose.